Added missing #includes on OpenBSD

Ronaldr1985 opened the pull request:
Resolves issues with compiling Odin on OpenBSD, sys/resource.h and sys/wait.h are required for WIFEXITED and WEXITSTATUS.

semarie commented on src/common.cpp:
```diff
@@ -25,6 +25,11 @@
 
+#if defined(GB_SYSTEM_OPENBSD)
+#include <sys/resource.h>
+#include <sys/wait.h>
```
only `sys/wait.h` should be required for `WIFEXITED` and `WEXITSTATUS` macro.

also, I would put the missing includes inside `src/gb/gb.h` under the `#if defined(GB_SYSTEM_OPENBSD)`

semarie commented on build_odin.sh:
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
I agree that it is the more portable way to invoke `bash`

Ronaldr1985 replied:
Yes, this means that the script can still run on OpenBSD without requiring changes to the SHABANG.

semarie commented:
@Ronaldr1985 ping ?

Ronaldr1985 commented:
Apologies for the large delay, I was busy and then I got sick unfortunately.

However, I have moved the include to gb/gb.h and removed it from common.cpp.
